cmd/sync-tags/main.go

```diff
@@ -199,15 +206,20 @@ func main() {
 			continue
 		}
 
-		// skip if it already exists in origin
-		if _, found := bTagCommits[bName]; found {
+		// if semver and non-semver tags need to be published, skip if both tags already exist
+		// if only non-semver tags need to be published, skip if the non-semver tag already exists
+		_, nonSemverTagExists := bTagCommits[bName]
+		_, semverTagExists := bTagCommits[semverTag]
+		if *publishSemverTags && semverTagExists && nonSemverTagExists {
+			continue
+		} else if nonSemverTagExists {
 			continue
 		}
 
 		// do not override tags (we build master first, i.e. the x.y.z-alpha.0 tag on master will not be created for feature branches)
-		if tagExists(r, bName) {
-			continue
-		}
+		// if tagExists(r, bName) {
+		//	continue
```

Comment:
@sttts question about this if statement: why is this required? don't we do the same check in:

https://github.com/kubernetes/publishing-bot/blob/dc65f29519695e122f8f4aa7dc1269ab6997884a/cmd/sync-tags/main.go#L202-L205

Reply:
wasn't the bName check above for the origin tag only? I think we rewrite tag names on fetch to `origin/<name>` or something like that. So we have local tags and origin tags. This check is for local tags, the upper for origin.

Reply:
Ah right, will update.